lit_tests/kernel/wave/mlir_converter_debug_locations.py

```diff
@@ -0,0 +1,257 @@
+    tkw.HardwareConstraint(threads_per_wave=64, vector_shapes={M: BLOCK_M, N: BLOCK_N}),
+]
+
+
+@wave.wave(constraints)
+def matrix_add(
+    # defines matrix in memory of req dimension with specific data types
+    a: Memory[M, N, ADDRESS_SPACE_A, tkl.f16],
+    b: Memory[M, N, ADDRESS_SPACE_B, tkl.f16],
+    c: Memory[M, N, ADDRESS_SPACE_C, tkl.f16],
+):
+    # Initialize the accumulator register with zeroes
+    c_reg = Register[M, N, tkl.f16](0.0)
+
+    # loads values from memory into registers
+    a_reg = wave.read(a)
+    b_reg = wave.read(b)
+
+    # compute the sum
+    c_reg = a_reg + b_reg
+
+    # writing results back to memory
+    wave.write(c_reg, c)
+
+
+@run_test
+def mlir_converter_location():
+    """Test MLIR converter debug location emission with a matrix addition kernel."""
+    # Set parameters for compilation
+    subs: dict[str | IndexSymbol, Any] = {
+        ADDRESS_SPACE_A: GLOBAL_ADDRESS_SPACE,
+        ADDRESS_SPACE_B: GLOBAL_ADDRESS_SPACE,
+        ADDRESS_SPACE_C: GLOBAL_ADDRESS_SPACE,
+        BLOCK_M: 64,
+        BLOCK_N: 64,
+        M: 128,
+        N: 128,
+    }
+
+    # Compile the kernel to get the trace
+    options = WaveCompileOptions(
+        subs=subs,
+        compile_to_mlir=True,  # Avoid IREE compilation
+        location_capture_config=LocationCaptureConfig(
+            level=LocationCaptureLevel.FILE_LINE_COL
+        ),
+    )
+    options = set_default_run_config(options)
+
+    # Compile the kernel to get the trace
+    compiled_kernel = wave_compile(options, matrix_add)
+
+    # Get the compiled graph from the compiled kernel
+    trace = compiled_kernel.get_compiled_graph()
+
+    constraints = matrix_add.constraints
+
+    # Use the mlir_converter to emit wave MLIR dialect
+    mlir_output, _ = emit_wave_dialect(trace, constraints, options, False)
+
+    # Print to stdout for FileCheck
+    print(mlir_output)
+
+    # CHECK-LABEL: mlir_converter_location
+    # CHECK: #loc = loc("{{.*}}mlir_converter_debug_locations.py":44:0)
+    # CHECK: module
+    # CHECK: func.func @kernel(%arg0: !wave.tensor<[@M, @N] of f16> loc("{{.*}}mlir_converter_debug_locations.py":44:0), %arg1: !wave.tensor<[@M, @N] of f16> loc("{{.*}}mlir_converter_debug_locations.py":44:0), %arg2: !wave.tensor<[@M, @N] of f16> loc("{{.*}}mlir_converter_debug_locations.py":44:0))
+
+    # CHECK: wave.read
+    # CHECK-SAME: loc(#loc1)
+
+    # CHECK: wave.read
+    # CHECK-SAME: loc(#loc2)
+
+    # CHECK: wave.add
+    # CHECK-SAME: loc(#loc3)
+
+    # CHECK: wave.write
+    # CHECK-SAME: loc(#loc4)
+
+    # CHECK: return loc(#loc)
+
+    # CHECK: loc(#loc)
+    # CHECK: loc(#loc)
+
+    # CHECK: #loc1 = loc("{{.*}}mlir_converter_debug_locations.py":55:12 to :24)
+    # CHECK: #loc2 = loc("{{.*}}mlir_converter_debug_locations.py":56:12 to :24)
+    # CHECK: #loc3 = loc("{{.*}}mlir_converter_debug_locations.py":59:12 to :25)
+    # CHECK: #loc4 = loc("{{.*}}mlir_converter_debug_locations.py":62:4 to :24)
```

Comment:
FYI, I remember that column capture is not supported prior to _some_ version of Python. We may only be running water tests on a single version, but let's structure the test defensively against that. IMO, it's fine not to check for exact columns here.